### PR TITLE
added gameoflife.java

### PR DESCRIPTION
**Pull request by rtaub5**

### CODE REVIEW

File: src/main/java/org/example/GOLBox.java [hunk 1]
```diff
@@ -0,0 +1,246 @@
+package org.example;
```
Comment (andrewoid) [inline src/main/java/org/example/GOLBox.java]: `taub.gameoflife` should be the package name

File: src/main/java/org/example/GameOfLife.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+
+    public int[][] getGameBoard()
+    {
+        return gameBoard;
+    }
+
+    public void nextGeneration()
+    {
+        for (int row = 0; row < gameBoard.length; ++row)
```
Comment (andrewoid) [inline src/main/java/org/example/GameOfLife.java]: `row++`

File: src/main/java/org/example/GameOfLife.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        {
+            for (int col = 0; col < gameBoard[row].length; ++col)
```
Comment (andrewoid) [inline src/main/java/org/example/GameOfLife.java]: `col++`

File: src/main/java/org/example/GameOfLife.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+            {
+                determineLiveOrDead(new GOLBox(this, row, col));
```
Comment (andrewoid) [inline src/main/java/org/example/GameOfLife.java]: // none edges
neighbors = isAlive(row+1, col+1) + isAlive(row+1, col) + isAlive(row+1, col-1)+ ... 

isAlive(row, col) returns 0 or 1
if (row < 0) return 0
if (col < 0) return 0 
if (row > gameBoard.length) return 0
if (row > gameBoard[0].length) return 0
return gameBoard[row][col]

File: src/main/java/taub/gameoflife/GameOfLife.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        origGameBoard = board;
+        setNewGameBoard();
+    }
+
+    private void setNewGameBoard()
+    {
+        newGameBoard = new int[origGameBoard.length][origGameBoard[0].length];
+        for (int i = 0; i < origGameBoard.length; i++)
+        {
+            newGameBoard[i] = origGameBoard[i].clone();
```
Comment (andrewoid) [inline src/main/java/taub/gameoflife/GameOfLife.java:21]: calling clone is not necessary

File: src/main/java/taub/gameoflife/Main.java [hunk 6]
```diff
@@ -8,15 +8,18 @@ public static void main(String[] args)
...
-            // for you, but you can always add more by pressing Cmd+F8.
-            System.out.println("i = " + i);
+            int [][] gameBoard = {
+                    {0, 0, 0},
+                    {1, 1, 1},
+                    {0, 0, 0}
+            };
+
+            GameOfLife game = new GameOfLife(gameBoard);
+            game.nextGeneration();
```
Comment (andrewoid) [inline src/main/java/taub/gameoflife/Main.java:20]: What happens if this is called a second time?

File: src/main/java/taub/gameoflife/GameOfLife.java (hunk 5 above)
Comment (rtaub5) [reply src/main/java/taub/gameoflife/GameOfLife.java:21]: If I take out .clone(), then it doesn't keep the arrays separate and my array returns like this: [0, 1, 1]
                            [1, 0, 1]
                            [0, 0, 0]

Comment (andrewoid) [reply src/main/java/taub/gameoflife/GameOfLife.java:21]: Nevermind

File: src/main/java/taub/gameoflife/GameOfLife.java [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+       int neighbors = isAlive(row - 1, col - 1) + isAlive(row - 1, col) + isAlive(row - 1, col + 1)
+                        + isAlive(row, col - 1) + isAlive(row, col + 1)
+                        + isAlive(row + 1, col - 1) + isAlive(row + 1, col) + isAlive(row + 1, col + 1);
+       if (neighbors == 3)
+       {
+           newGameBoard[row][col] = 1;
+       } else if (neighbors != 2)
+       {
+           newGameBoard[row][col] = 0;
+       }
```
Comment (andrewoid) [inline src/main/java/taub/gameoflife/GameOfLife.java:58]: If the cell is alive in origGameBoard and neighbors==2, will it be alive in newGameBoard?

Comment (rtaub5) [reply src/main/java/taub/gameoflife/GameOfLife.java:58]: Yes, it'll just stay a 1. newGameBoard starts off as an exact copy or origGameBoard but then it changes based on the Neighbors.

**Review by andrewoid — APPROVED** [review]

**Pull request merged**